apps/laboratory/src/pages/library/ethers-siwe.tsx

```diff
@@ -5,26 +5,28 @@ import { createWeb3Modal, defaultConfig } from '@web3modal/ethers5/react'
 import { ThemeStore } from '../../utils/StoreUtil'
 import { EthersConstants } from '../../utils/EthersConstants'
 import { ConstantsUtil } from '../../utils/ConstantsUtil'
-import { siweConfig } from '../../utils/SiweUtils'
-
-const modal = createWeb3Modal({
-  ethersConfig: defaultConfig({
-    metadata: ConstantsUtil.Metadata,
-    defaultChainId: 1,
-    rpcUrl: 'https://cloudflare-eth.com'
-  }),
-  chains: EthersConstants.chains,
-  projectId: ConstantsUtil.ProjectId,
-  enableAnalytics: true,
-  metadata: ConstantsUtil.Metadata,
-  siweConfig,
-  enableOnramp: true,
-  customWallets: ConstantsUtil.CustomWallets
-})
-
-ThemeStore.setModal(modal)
+import { useMemo } from 'react'
 
 export default function EthersSiwe() {
+  useMemo(async () => {
+    const { siweConfig } = await import('../../utils/SiweUtils')
+    const modal = createWeb3Modal({
+      ethersConfig: defaultConfig({
+        metadata: ConstantsUtil.Metadata,
+        defaultChainId: 1,
+        rpcUrl: 'https://cloudflare-eth.com'
+      }),
+      chains: EthersConstants.chains,
+      projectId: ConstantsUtil.ProjectId,
+      enableAnalytics: true,
+      metadata: ConstantsUtil.Metadata,
+      siweConfig,
+      enableOnramp: true,
+      customWallets: ConstantsUtil.CustomWallets
+    })
+    ThemeStore.setModal(modal)
+  }, [])
+
```

Comment:
Curious why are we moving this into the component ?

Reply:
https://github.com/WalletConnect/web3modal/blob/435505ecc37ba3d06aeb3a34bdb03972e233e8d2/apps/laboratory/src/utils/SiweUtils.ts#L9

results in `Window is not defined` Next.js exception so we have to execute this only client side